server/src/main/java/org/elasticsearch/health/HealthIndicatorResult.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.health;
+
+import org.elasticsearch.xcontent.ToXContentFragment;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+public record HealthIndicatorResult(String name, String component, HealthStatus status, String summary, ToXContentFragment details)
```

Comment:
I think details should no longer be a fragment. Ideally we would add a little test indicator to flesh out such things, but we can defer that to a follow-up.

Reply:
Will follow up as soon as this is merged :+1:

Reply:
Followup pr: https://github.com/elastic/elasticsearch/pull/83417